main/main.c

```diff
@@ -1327,38 +1359,19 @@ static ZEND_COLD void php_error_cb(int orig_type, const char *error_filename, co
 		}
 
 		if (PG(display_errors) && ((module_initialized && !PG(during_request_startup)) || (PG(display_startup_errors)))) {
-			if (PG(xmlrpc_errors)) {
-				php_printf("<?xml version=\"1.0\"?><methodResponse><fault><value><struct><member><name>faultCode</name><value><int>" ZEND_LONG_FMT "</int></value></member><member><name>faultString</name><value><string>%s:%s in %s on line %" PRIu32 "</string></value></member></struct></value></fault></methodResponse>", PG(xmlrpc_error_number), error_type_str, buffer, error_filename, error_lineno);
-			} else {
+			if (zend_error_display_handle(type, error_filename, error_lineno, buffer, buffer_len) == 0) {
+				// default error handling if no display callback was triggered
```

Comment:
Shouldn't there always be at least the default callback?

Reply:
Nothing wrong with defensive coding :-) (Instead of `== 0` at the end, I would have use `if (!zend_error_display_handle...`.